src/box/txn_limbo.c

```diff
@@ -285,8 +285,11 @@ txn_limbo_wait_complete(struct txn_limbo *limbo, struct txn_limbo_entry *entry)
 			goto complete;
 		if (txn_limbo_is_frozen(limbo))
 			goto wait;
-		if (rc != 0)
+		if (rc != 0) {
+			if (fiber_is_cancelled())
```

Comment:
The check for `fiber_is_cancelled` should go before the `txn_limbo_is_frozen()` check now. Otherwise a cancelled fiber will continue running until the limbo is unfrozen.

Reply:
Done. Changed to:

```diff
@@ -283,6 +283,8 @@ txn_limbo_wait_complete(struct txn_limbo *limbo, struct txn_limbo>
                int rc = fiber_cond_wait_timeout(&limbo->wait_cond, timeout);
                if (txn_limbo_entry_is_complete(entry))
                        goto complete;
+               if (rc != 0 && fiber_is_cancelled())
+                       return -1;
                if (txn_limbo_is_frozen(limbo))
                        goto wait;
                if (rc != 0)
```